hamocc/mo_parambgc_ini.F90

```diff
@@ -0,0 +1,1241 @@
+    !     Dust deposition and iron solubility parameters
+    !********************************************************************
+    !ik weight percent iron in dust deposition times Fe solubility
+    ! the latter three values come from Johnson et al., 1997
+    fetune     = 0.6                  ! factor introduced to tune deposition/solubility
+    perc_diron = fetune * 0.035 * 0.01 / 55.85
+    fesoly     = 0.5*1.e-9            ! max. diss. iron concentration in deep water 
+    relaxfe    = 0.05/365.            ! 1/d complexation rate to relax iron concentration to fesoly
+
+    !********************************************************************
+    !     Sinking parameters
+    !********************************************************************
+    wpoc   =  5.       !m/d   Sinking speed of detritus iris : 5.
+    wcal   = 30.       !m/d   Sinking speed of CaCO3 shell material
+    wopal  = 30.       !m/d   Sinking speed of opal iris : 60
+#if defined(WLIN) && ! defined(AGG)
+    wmin   =  1.       !m/d   minimum sinking speed
+    wmax   = 60.       !m/d   maximum sinking speed
+    wlin   = 60./2400. !m/d/m constant describing incr. with depth, r/a=1.0
+#endif
+#ifndef AGG
+    dustd1   = 0.0001 !cm = 1 um, boundary between clay and silt
+    dustd2   = dustd1*dustd1
+    dustsink = (9.81 * 86400. / 18.                  &  ! g * sec per day / 18.
+     &         * (claydens - 1025.) / 1.567 * 1000.    &  !excess density / dyn. visc.
+     &         * dustd2 * 1.e-4)
+    wdust = dustsink
+#endif    
+  end subroutine
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine read_bgcnamelist()
+    !
+    ! Read the bgcparams namelist for parameter tuning.
+    ! Note that afterward, i) rates need to be adjusted for timestep
+    ! and some depending parameters need re-calculation
+    !
+
+    integer  :: iounit
+
+    namelist /bgcparams/ bkphy,dyphy,bluefix,bkzoo,grazra,spemor,gammap,gammaz,ecan,zinges,epsher,bkopal,rcalc,ropal,              &
+                       & remido,drempoc,dremopal,dremn2o,dremsul,fetune,relaxfe,wpoc,wcal,wopal
+
+    open (newunit=iounit, file=bgc_namelist, status='old',action='read')
+    read (unit=iounit, nml=BGCPARAMS)
+    close (unit=iounit)
+
+    if (mnproc.eq.1) then
+      write(io_stdo_bgc,*) '------------------------------------------'
+      write(io_stdo_bgc,*) 'iHAMOCC: read namelist bgcparams'
+      write(io_stdo_bgc,nml=BGCPARAMS)
+      write(io_stdo_bgc,*) '------------------------------------------'
+    endif
+
+  end subroutine 
+ 
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine readjust_param()
+    !
+    !  AFTER reading the namelist:
+    ! re-adjust parameters that depend on tuning parameters
+    !
+    perc_diron = fetune * 0.035 * 0.01 / 55.85
+
+  end subroutine
```

Comment:
If this remains the only parameter to adjust, I think it is better to just have this line in the main routine instead of making a separate subroutine.

Reply:
I am not sure, if I understand, what you consider the main routine here. I guess, my intention was to provide a structure , which one can follow.